daemon/container_operations.go

```diff
@@ -947,78 +947,56 @@ func errRemovalContainer(containerID string) error {
 }
 
 // ConnectToNetwork connects a container to a network
-func (daemon *Daemon) ConnectToNetwork(container *container.Container, idOrName string, endpointConfig *networktypes.EndpointSettings) error {
+func (daemon *Daemon) ConnectToNetwork(container *container.Container, network libnetwork.Network, endpointConfig *networktypes.EndpointSettings) error {
 	if endpointConfig == nil {
 		endpointConfig = &networktypes.EndpointSettings{}
 	}
 	if !container.Running {
 		if container.RemovalInProgress || container.Dead {
 			return errRemovalContainer(container.ID)
 		}
-
-		n, err := daemon.FindNetwork(idOrName)
-		if err == nil && n != nil {
-			if err := daemon.updateNetworkConfig(container, n, endpointConfig, true); err != nil {
-				return err
-			}
-		} else {
-			container.NetworkSettings.Networks[idOrName] = &network.EndpointSettings{
```

Comment:
Yes, for the connect to a non-running container not erroring out  issue.
Before https://github.com/docker/docker/pull/25962/commits/99a98ccc14a9427be47c8006e130750710db0a16, the network existence check was deferred to `daemon.updateNetworkConfig()`